firestore/google/cloud/firestore_v1beta1/watch.py

```diff
@@ -396,7 +393,7 @@ def _on_snapshot_target_change_remove(self, proto):
 
         # TODO: Consider surfacing a code property on the exception.
         # TODO: Consider a more exact exception
-        raise Exception("Error %s:  %s" % (code, message))
+        raise AssertionError("Error %s:  %s" % (code, message))
```

Comment:
We prefer to avoid `AssertionError` in non-testing code.  Maybe `RuntimeError`?  Or would be we be better off raising the exception returned by `google.api_core.exceptions.exception_class_for_grpc_status`?

Reply:
Yeah.  I used AssertionError because there are a bunch of assert() statements in various methods in watch.py and it's unclear which methods should raise which kind of error if we change this one case.  I guess we could use some existing api_core.exceptions exception, but those are currently all square peg in round hole because they are raised when the client is calling the server via HTTP, and watch is the inverse case.  Maybe I should change api_core.exceptions to have some generic WatchError that inherits from GoogleAPICallError, but even GoogleAPICallError is square peg because the errors dont happen during any API call.  Maybe just a custom exception class in watch.py itself.

Reply:
I changed it to RuntimeError if only to not have to document a one-off exception type.